bridges/ArtStationBridge.php

```diff
@@ -0,0 +1,84 @@
+<?php
+class ArtStationBridge extends BridgeAbstract {
+	const NAME = 'ArtStation';
+	const URI = 'https://www.artstation.com';
+	const DESCRIPTION = 'Fetches the latest ten artworks from a search query on ArtStation.';
+	const MAINTAINER = 'thefranke';
+	const CACHE_TIMEOUT = 3600; // 1h
+
+	const PARAMETERS = array(
+		'Search Query' => array(
+			'q' => array(
+				'name' => 'Search term',
+				'required' => true
+			)
+		)
+	);
+
+	public function getIcon() {
+		return 'https://www.artstation.com/assets/favicon-58653022bc38c1905ac7aa1b10bffa6b.ico';
+	}
+
+	public function getName() {
+		return self::NAME . ': ' . $this->getInput('q');
+	}
+
+	public function fetchSearch($searchQuery) {
+		$data = '{"query":"' . $searchQuery . '","page":1,"per_page":50,"sorting":"date",';
+		$data .= '"pro_first":"1","filters":[],"additional_fields":[]}';
+
+		$header = array(
+			'Content-Type: application/json',
+			'Accept: application/json'
+		);
+
+		$opts = array(
+			CURLOPT_POST => true,
+			CURLOPT_POSTFIELDS => $data,
+			CURLOPT_RETURNTRANSFER => true
+		);
+
+		$jsonSearchURL = self::URI . '/api/v2/search/projects.json';
+		$jsonSearchStr = getContents($jsonSearchURL, $header, $opts) or die('Could not fetch JSON for search query.');
+		return json_decode($jsonSearchStr);
+	}
+
+	public function fetchProject($hashID) {
+		$jsonProjectURL = self::URI . '/projects/' . $hashID . '.json';
+		$jsonProjectStr = getContents($jsonProjectURL) or die('Could not fetch JSON for project.');
+		return json_decode($jsonProjectStr);
+	}
+
+	public function collectData() {
+		$searchTerm = $this->getInput('q');
+		$jsonQuery = $this->fetchSearch($searchTerm);
+
+		foreach($jsonQuery->data as $media) {
+			// get detailed info about media item
+			$jsonProject = $this->fetchProject($media->hash_id);
+
+			// create item
+			$item = array();
+			$item['title'] = $media->title;
+			$item['uri'] = $media->url;
+			$item['timestamp'] = strtotime($jsonProject->published_at);
+			$item['author'] = $media->user->full_name;
+			$item['categories'] = implode(',', $jsonProject->tags);
+
+			$item['content'] = '<a href="' . $media->url . '">
+								<img style="max-width: 100%" src="' . $jsonProject->cover_url . '">
+								</a>' . $jsonProject->description;
+
```

Comment:
The lines above look strange.

```php
$item['content'] = '<a href="'
. $media->url
. '"><img style="max-width: 100%" src="'
. $jsonProject->cover_url
. '"></a>'
. $jsonProject->description;
```

I know this is not perfect, but this is all you get with concatenation. Alternatively make use of variables like so:

```php
$item['content'] = <<<EOD
<a href="{$media->url}">
	<img style="max-width: 100%" src="{$jsonProject->cover_url}">
</a>{$jsonProject->description}
EOD;
```

Please note, that "EOD;" must not be indented for this to work (this also means, anything between "<<<EOD" and "EOD;" doesn't need to be indented).

I haven't tested it, but this should work just fine.

Reply:
I'm struggling a bit with the Heredoc version, so I'll stick with concatenation.